graph-commons/src/main/scala/io/renku/events/producers/EventSender.scala

```diff
@@ -48,8 +49,19 @@ trait EventSender[F[_]] {
   ): F[Unit]
 }
 
+object EventSender {
+  def apply[F[_]: Async: Logger: MetricsRegistry](
+      consumerUrl: EventConsumerUrl
+  ): F[EventSender[F]] = for {
+    serviceUrl      <- consumerUrl()
+    sentEventsGauge <- SentEventsGauge[F]
+  } yield new EventSenderImpl(serviceUrl, sentEventsGauge, onErrorSleep = 15 seconds)
+
+  final case class EventContext(categoryName: CategoryName, errorMessage: String)
+}
+
 class EventSenderImpl[F[_]: Async: Logger](
-    eventLogUrl:            EventLogUrl,
+    serviceUrl:             UrlTinyType,
```

Comment:
I was wondering why not use the new trait `EventConsumerUrl` as argument type? Can't see it used anywhere (but I probably missed it).